src/hooks/useVideoMetadata.ts

```diff
@@ -0,0 +1,42 @@
+import { useEffect, useState } from 'react'
+
+export type VideoMetaData = {
+  type: string
+  sizeInBytes: number
+  height: number
+  width: number
+  duration: number
+}
+export const useVideoMetadata = () => {
+  const [file, setFile] = useState<File>()
+  const [metadata, setMetadata] = useState<VideoMetaData>()
+  useEffect(() => {
```

Comment:
I'm thinking that maybe we could actually make this synchronous? Right now you need to pass the file to the hook and possibly wait a render to get the data back. This will probably make some flows harder, for example when we want to validate the file (minimum resolution) in the upload flow. Maybe this doesn't have to be a hook but just an async function?

Reply:
The problem is we need to wait for the listener loadeddata as the video doesn't get processed inmediatly